### PR TITLE
20171220 move to c++11

### PR DESCRIPTION
**Pull request by hjmjohnson**
@malaterre Just a discussion starter.  PLEASE DO NOT COMMIT.

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@malaterre I would like to know your thoughts about agressively moving GDCM to C++11 syntax.  VTK has already made the switch, and ITK is well on the way to doing a similar conversion.

**Comment by hjmjohnson** [comment]
@malaterre Do you have any suggestions on this set of code changes?

**Comment by malaterre** [comment]
@hjmjohnson Do you have a suggestion for the build failure ?

**Comment by malaterre** [comment]
@hjmjohnson Sorry for the CharLS mess. I believe this PR should be ready to go now. Please resolve conflict + rebase. Thanks

**Comment by hjmjohnson** [comment]
@malaterre Rebased.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,21 +1,9 @@
...
-if(POLICY CMP0026)
-  cmake_policy(SET CMP0026 NEW)
-endif()
-if(POLICY CMP0042)
-  cmake_policy(SET CMP0042 NEW)
-endif()
-if( POLICY CMP0063 )
-  cmake_policy(SET CMP0063 NEW)
-endif()
+cmake_minimum_required(VERSION 3.8.2 FATAL_ERROR)
```
Comment (malaterre) [inline CMakeLists.txt]: This is a no-go for me. I use Travis to build GDCM, currently only Ubuntu Trusty image are available, cmake version is 2.8.12

**Comment by malaterre** [comment]
@hjmjohnson I've fixed the build issue on appveyor + travis, please rebase one last time. Thanks

**Comment by hjmjohnson** [comment]
@malaterre Yeah!  All checks passed!

**Comment by malaterre** [comment]
@hjmjohnson Thanks for your work (and patience) !

**Pull request merged**